### PR TITLE
Link to reference for deployment properties

### PR DESCRIPTION
**Pull request by dturanski**
Addresses #259.  For consistency, it is better to update the SCDF reference guide and link to that.

### CODE REVIEW

File: content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md [hunk 1]
```diff
@@ -11,7 +11,18 @@ When deploying a stream, properties fall into two groups:
 - Properties that control how the apps are deployed to the target platform and that use a `deployer` prefix are referred to as _deployer properties_.
 - Properties that control or override how the application behave and that are set during stream creation are referred to as _application properties_.
 
-You need to pick a defined platform configuration where each platform type (`local`, `cloudfoundry` or `kubernetes`) has a different set of possible deployment properties. Every platform has a set of generic properties for `memory`, `cpu`, and `disk` reservations and `count` to define how many instances should be created on that platform. The following image shows the Deploy Stream Definition view, where you can set these properties:
+You need to pick a defined platform configuration where each platform type (`local`, `cloudfoundry` or `kubernetes`) has a different set of possible deployment properties. Every platform has a set of generic properties for `memory`, `cpu`, and `disk` reservations and `count` to define how many instances should be created on that platform.
```
Comment (Buzzardo) [inline content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md:14]: Add a comma after `cloudfoundry`.

File: content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md [hunk 2]
```diff
@@ -11,7 +11,18 @@ When deploying a stream, properties fall into two groups:
...
+
+<!--TIP-->
+
+You can view the deployment properties for each of the platforms by selecting one of the following links: [local](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#configuration-local-deployer), [cloudfoundry](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#configuration-cloudfoundry-deployer) or, [kubernetes](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#configuration-kubernetes-deployer).
```
Comment (Buzzardo) [inline content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md:18]: Move the comma after "or" to before "or".

File: content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md [hunk 3]
```diff
@@ -11,7 +11,18 @@ When deploying a stream, properties fall into two groups:
...
+
+<!--END_TIP-->
+
+The following image shows the Deploy Stream Definition view, where you can set these properties:
```
Comment (Buzzardo) [inline content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md:22]: The next four lines should be:

You can set the following properties:

* Deployer Properties: These properties customize how tasks are launched.
* Application Properties: These are application specific properties.

The following image shows the Deploy Stream Definition view:

File: content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md [hunk 4]
```diff
@@ -52,3 +63,12 @@ The preceding example would be as follows in SCDF shell:
 stream deploy --name ticktock --properties "app.time.trigger.initial-delay=1,deployer.*.cpu=1,deployer.*.local.shutdown-timeout=60,deployer.*.memory=512,deployer.log.count=2,deployer.log.local.delete-files-on-exit=false,deployer.time.disk=512,spring.cloud.dataflow.skipper.platformName=local-debug"
 Deployment request has been sent for stream 'ticktock'
 ```
+
+## Platform Specific Deployer Properties
+
+### Cloud Foundry Deployer Properties
+
+The Cloud Foundry Deployer maps task and application properties to environment variable `SPRING_APPLICATION_JSON` by default. This is defined in the generated Cloud Foundry application manifest.
```
Comment (Buzzardo) [inline content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md:71]: Change the first sentence to: The Cloud Foundry Deployer maps task and application properties to an environment variable (by default, `SPRING_APPLICATION_JSON`).

File: content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md [hunk 5]
```diff
@@ -52,3 +63,12 @@ The preceding example would be as follows in SCDF shell:
...
+The value is a JSON document and is a standard Spring Boot property source.
+You can optionally configure the deployer to create top level environment variables by setting `deployer.<app>.cloudfoundry.use-spring-application-json=false`.
```
Comment (Buzzardo) [inline content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md:73]: Hyphenate "top level" (so "top-level").

File: content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md [hunk 6]
```diff
@@ -52,3 +63,12 @@ The preceding example would be as follows in SCDF shell:
...
+You may also add top-level environment variables explicitly using `deployer.<app>.cloudfoundry.env.<key>=<value>`. This is useful for adding [Java build pack configuration properties](https://github.com/cloudfoundry/java-buildpack) to the application manifest since the Java build pack applies its properties before the application starts and does not treat `SPRING_APPLICATION_JSON` as a special case.
```
Comment (Buzzardo) [inline content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md:74]: Add "by" before "using".

File: content/documentation/pages/5-feature-guides/3-batch/1-deployment-properties.md [hunk 7]
```diff
@@ -6,12 +6,18 @@ description: 'Initiate a Batch deployment with deployment property overrides'
 
 # Deployment Properties
 
-When task definitions are launched to the target platforms `Local`, `CloudFoundry` and `Kubernetes`, you can provide the configuration properties that are applied to the task applications at launch time.
+When task definitions are launched to the target platforms `local`, `cloudFoundry` and `kubernetes`, you can provide the configuration properties that are applied to the task applications at launch time.
```
Comment (Buzzardo) [inline content/documentation/pages/5-feature-guides/3-batch/1-deployment-properties.md:9]: Change this sentence to:

When task definitions are launched to the target platforms (`local`, `cloudFoundry`, and `kubernetes`), you can provide the configuration properties that are applied to the task applications at launch time.

(The difference is the parentheses around the list and the serial comma in the list.)

File: content/documentation/pages/5-feature-guides/3-batch/1-deployment-properties.md [hunk 8]
```diff
@@ -6,12 +6,18 @@ description: 'Initiate a Batch deployment with deployment property overrides'
...
 For instance you can specify:
 
 - Deployer Properties - These properties customize how tasks are launched.
 - Application Properties - These are application specific properties.
 
+<!--TIP-->
+
+You can view the deployment properties for each of the platforms by selecting one of the following links: [local](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#configuration-local-deployer), [cloudfoundry](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#configuration-cloudfoundry-deployer) or, [kubernetes](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#configuration-kubernetes-deployer).
```
Comment (Buzzardo) [inline content/documentation/pages/5-feature-guides/3-batch/1-deployment-properties.md:17]: Move the comma after "or" to before "or".

File: content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md (hunk 3 above)
Comment (dturanski) [reply content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md:22]: Actually this content is a repeat of the start of the section, cut and paste from the corresponding page for tasks. I will clean up.

File: content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md (hunk 4 above)
Comment (dturanski) [reply content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md:71]: This does not convey the same meaning to me.  `SPRING_APPLICATION_JSON` is a special boot variable. The name is not a default setting.  The alternative configuration maps each property to a separate env variable.

Comment (dturanski) [reply content/documentation/pages/5-feature-guides/2-streams/1-deployment-properties.md:71]: But a good catch anyway. I meant to remove this section.

**Pull request merged**